### PR TITLE
Add abort UI to Prow UI and Spyglass.

### PR DESCRIPTION
**Pull request by carolinar7**
This PR adds the interface to users to be able to abort a ProwJob on the Prow UI and Spyglass. The abort button will be disabled whenever a Job is not in a Triggered, or Pending state.

Screenshot 2022-07-19 2 11 41 PM
image
image

Fixes: https://github.com/kubernetes/test-infra/issues/24390

/cc @chaodaiG @cjwagner @e-blackwelder

### CODE REVIEW

File: prow/cmd/deck/static/common/abort.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import {showAlert, showToast, State} from "./common";
+
+export function createAbortProwJobIcon(state: string, prowjob: string, csrfToken: string): HTMLElement {
+  const url = `${location.protocol}//${location.host}/abort?prowjob=${prowjob}`;
+  const button = document.createElement('button');
+  button.classList.add('mdl-button', 'mdl-js-button', 'mdl-button--icon');
+  button.innerHTML = '<i class="icon-button material-icons" title="Cancel this job" style="color: gray">cancel</i>';
+
+  if (state !== State.TRIGGERED && state !== State.PENDING) {
+    button.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
```
Comment (e-blackwelder) [inline prow/cmd/deck/static/common/abort.ts:10]: I'd be _very_ careful adding anything into literal HTML that was passed as a simple string via method parameter.

I like to ensure my code works/is safe/etc when viewed "in isolation". Meaning if I _only_ review this one method, I can't ensure that the `state` being passed in isn't some user-provided string.

I think one solution would be to change `string` type to the `enum` type that I think you're _actually_ using somewhere... then I'd know that it's only one of _N_ possibilities and I can navigate directly to those from my IDE. :)

**Review by e-blackwelder — COMMENTED** [review]
I like it.

Code looks great, in general.

One security warning that I think would be good to address.

File: prow/cmd/deck/static/common/abort.ts (hunk 1 above)
Comment (carolinar7) [reply prow/cmd/deck/static/common/abort.ts:11]: This has been updated! It uses another type, `ProwJobState`, that is defined in the prow api (typescript).

**Review by e-blackwelder — APPROVED** [review]
Looks good to me.

**Comment by chaodaiG** [comment]
/hold
I think we should have a pop-up to prevent accidentally one touch aborting

**Comment by carolinar7** [comment]
@e-blackwelder Just updated this PR to also contain a confirmation modal! View the latest commit for changes.

File: prow/cmd/deck/static/common/abort.ts [hunk 2]
```diff
@@ -1,41 +1,76 @@
...
+    if (event.target === modal) {
+      closeModal();
+    }
+  };
   if (state !== State.TRIGGERED && state !== State.PENDING) {
-    button.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
-    button.classList.remove('icon-button');
-    button.disabled = true;
+    abortButton.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
+    abortButton.classList.remove('icon-button');
```
Comment (e-blackwelder) [inline prow/cmd/deck/static/common/abort.ts:28]: Does it ever have that class? It only looks like the nested `<i>` inside does.

**Review by e-blackwelder — APPROVED** [review]
Looks reasonable. But is beginning to push the limits of what I'm comfortable doing with manual DOM manipulations... A more systematic approach (i.e. component architecture) would be better, but _waaaay_ outside the scope of this work. Nicely done.

File: prow/cmd/deck/static/common/abort.ts (hunk 2 above)
Comment (carolinar7) [reply prow/cmd/deck/static/common/abort.ts]: Good catch there, you are right! This has been fixed!

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**